### PR TITLE
Adding the Fourier correlation sample

### PR DESCRIPTION
**Pull request by hagabb**
Signed-off-by: hagabb <henry.a.gabb@intel.com>

# Description

This pull request is to add a Fourier correlation sample to the Libraries/oneMKL section of the oneAPI-samples repository. The sample consists of two new codes: one that implements a 1D Fourier correlation using explicit buffering and another that implements it with USM. The latter also includes a maxloc reduction. The sample shows how to compose more complicated mathematical operations using multiple oneMKL kernel functions while paying attention to host-device data placement and data transfer.

Fixes # (issue) N/A

## Type of change

Please delete options that are not relevant. Add a 'X' to the one that is applicable. 

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Sample Migration (Moving sample from old repository after completing checklist established)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Command Line
- [ ] oneapi-cli
- [ ] Visual Studio
- [ ] Eclipse IDE
- [ ] VSCode

# Checklist for Moving samples:
Links and Details can be found in the samples WG Teams Files. 

- [ ] Review sample design with domain reviewers https://github.com/oneapi-src/oneAPI-samples/wiki/Reviewers-and-Domain-Experts 
- [ ] Implement coding guidelines and ensure code quality.
- [ ] Adhere to sample.json specification. https://github.com/oneapi-src/oneAPI-samples/wiki/sample-json-specification
- [ ] Run jsonlint on sample.json to verify json syntax. www.jsonlint.com
- [ ] Adhere to readme template 
- [ ] Ensure/create CI test configurations for sample (ciTests field) https://github.com/oneapi-src/oneAPI-samples/wiki/sample-json-ci-test-object
- [ ] Enforce format via clang-format config file
- [ ] Review DPC++ code with Paul Peterseon. (GitHub User: pmpeter1)
- [ ] Review readme with Tom Lenth or Joe Oster. (GitHub User: JoeOster)
- [ ] Tested using Dev Cloud when applicable
- [ ] Implement fixes for ONSAM Jiras
- [ ] If you have new dependencies/binaries, inform Samples Project Manager Swapna R Dontharaju (@srdontha) or @JoeOster

### CODE REVIEW

**Review by JoeOster — CHANGES_REQUESTED** [review]
Please update the description of the PR

**Comment by JoeOster** [comment]
@pmpeter1 please take a look at this

File: Libraries/oneMKL/fourier_correlation/sample.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
+{
+  "guid": "69F6B0A1-3046-467F-B4A7-E61F636BC5BF",
+  "name": "Fourier Correlation",
```
Comment (JoeOster) [inline Libraries/oneMKL/fourier_correlation/sample.json:3]: I updated the readme and sample.json file, please inspect and verify I didn't change your message too much

**Review by JoeOster — CHANGES_REQUESTED** [review]

File: Libraries/oneMKL/fourier_correlation/sample.json (hunk 1 above)
Comment (hagabb) [reply Libraries/oneMKL/fourier_correlation/sample.json:3]: These changes and fixes are fine.

**Review by hagabb — COMMENTED** [review]
The changes and fixes made by @JoeOster are fine.

**Review by JoeOster — APPROVED** [review]

**Comment by JoeOster** [comment]
Waiting for comments from @pmpeter1

File: Libraries/oneMKL/fourier_correlation/sample.json [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    "linux": [
+      {
+        "id": "fourier_correlation",
+        "steps": [
+          "make clean",
+          "make"
+        ]
+      }
+    ],
+    "windows": [
```
Comment (JoeOster) [inline Libraries/oneMKL/fourier_correlation/sample.json:22]: The windows build is trowing errors, please take a look at the run for details

**Review by JoeOster — CHANGES_REQUESTED** [review]

File: Libraries/oneMKL/fourier_correlation/makefile [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+default: run_all
+
+all: run_all
+
+run_all: fcorr_1d_buff.exe fcorr_1d_usm.exe
+	.\fcorr_1d_buff 1024
+	.\fcorr_1d_usm 1024
+
+DPCPP_OPTS=-DMKL_ILP64  -I"%MKLROOT%\include" mkl_sycl_dll.lib mkl_intel_ilp64_dll.lib mkl_tbb_thread_dll.lib mkl_core_dll.lib tbb.lib sycl.lib OpenCL.lib
```
Comment (aelizaro) [inline Libraries/oneMKL/fourier_correlation/makefile:11]: Hi @hagabb, for windows mkl_sycl_dll.lib supports only blas and lapack domains, so you need to use static mkl_sycl.lib

**Review by JoeOster — APPROVED** [review]

**Pull request merged**